share/smack/lib/smack.c

```diff
@@ -42,6 +42,7 @@ void __VERIFIER_error(void) {
 }
 
 void exit(int x) {
+  __SMACK_code("assert $eq.ref.bool($allocatedCounter, 0);");
```

Comment:
You do not need to call `eq.ref.bool` function. Just say `assert $allocatedCounter == 0;`